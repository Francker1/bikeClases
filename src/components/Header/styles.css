.header {
  background-color: green;
}

.header .navlink.active {
  color: red;
}
