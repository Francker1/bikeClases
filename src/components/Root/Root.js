import React from 'react';
import T from 'prop-types';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

export default function Root({ store, children }) {
  return (
    <Provider store={store}>
      <Router>{children}</Router>
    </Provider>
  );
}

Root.propTypes = {
  store: T.shape({
    dispatch: T.func.isRequired,
    getState: T.func.isRequired,
    subscribe: T.func.isRequired,
  }).isRequired,
};
