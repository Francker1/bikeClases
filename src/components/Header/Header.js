import React from 'react';
import { NavLink } from 'react-router-dom';

import './styles.css';
import classNames from 'classnames';

const validFilters = ['/mountain', '/road'];

export default function Header({ className, cartItems }) {
  return (
    <header className={classNames('header', className)}>
      <h1 className="title">BIKES - STORE</h1>
      <div className="navbar">
        <NavLink
          className="navlink"
          isActive={({ isExact }, { pathname }) =>
            isExact || validFilters.includes(pathname)
          }
          to="/"
        >
          Store
        </NavLink>
        <NavLink className="navlink" exact to="/cart">
          {`Checkout (${cartItems})`}
        </NavLink>
      </div>
    </header>
  );
}
