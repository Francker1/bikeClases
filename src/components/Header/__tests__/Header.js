import React from 'react';
import { shallow, mount } from 'enzyme';

import Root from '../../Root';
import Header from '../Header';
import ConnectedHeader from '../../Header';

describe('Header', () => {
  const defaultProps = {
    totalCartItems: 12,
  };

  const render = props => shallow(<Header {...defaultProps} {...props} />);
  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });

  it('should render Header component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('first link should be active if match is exact', () => {
    const match = { isExact: true };
    const location = {};
    expect(
      wrapper
        .find('.link')
        .at(0)
        .props()
        .isActive(match, location),
    ).toBe(true);
  });

  it('first link should be active if pathname is valid', () => {
    const match = { isExact: false };
    const location = { pathname: '/road' };
    expect(
      wrapper
        .find('.link')
        .at(0)
        .props()
        .isActive(match, location),
    ).toBe(true);
  });
});

describe('ConnectedHeader', () => {
  const defaultProps = {};
  const store = {
    getState: () => ({
      cart: {
        bike1: 2,
      },
    }),
    dispatch: () => {},
    subscribe: () => {},
  };
  const render = props =>
    mount(
      <Root store={store}>
        <ConnectedHeader {...defaultProps} {...props} />{' '}
      </Root>,
    );

  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });

  it('should render connected Header component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
