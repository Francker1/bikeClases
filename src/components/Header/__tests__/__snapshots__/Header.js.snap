// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedHeader should render connected Header component 1`] = `
<Root
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Header)>
          <Header
            dispatch={[Function]}
            totalCartItems={2}
          >
            <header
              className="header"
            >
              <h1
                className="title"
              >
                BIKES - STORE
              </h1>
              <div
                className="navbar"
              >
                <NavLink
                  className="link"
                  isActive={[Function]}
                  to="/"
                >
                  <Link
                    aria-current="page"
                    className="link active"
                    style={Object {}}
                    to="/"
                  >
                    <a
                      aria-current="page"
                      className="link active"
                      href="/"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      Bikes
                    </a>
                  </Link>
                </NavLink>
                <NavLink
                  className="link"
                  exact={true}
                  to="/cart"
                >
                  <Link
                    aria-current={null}
                    className="link"
                    to="/cart"
                  >
                    <a
                      aria-current={null}
                      className="link"
                      href="/cart"
                      onClick={[Function]}
                    >
                      Checkout (2)
                    </a>
                  </Link>
                </NavLink>
              </div>
            </header>
          </Header>
        </Connect(Header)>
         
      </Router>
    </BrowserRouter>
  </Provider>
</Root>
`;

exports[`Header should render Header component 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    BIKES - STORE
  </h1>
  <div
    className="navbar"
  >
    <NavLink
      className="link"
      isActive={[Function]}
      to="/"
    >
      Bikes
    </NavLink>
    <NavLink
      className="link"
      exact={true}
      to="/cart"
    >
      Checkout (12)
    </NavLink>
  </div>
</header>
`;
